QuickHull.cpp

```diff
@@ -91,6 +91,31 @@ namespace quickhull {
 		return ConvexHull<T>(m_mesh,m_vertexData, CCW, useOriginalIndices);
 	}
 
+  // Check if the mesh remains convex after adding new faces
+  template<typename T>
+    bool QuickHull<T>::isMeshConvex(const std::vector<size_t>& newFaces) {
+      for (const size_t& faceIndex : newFaces) {
+        const auto& face = m_mesh.m_faces[faceIndex];
+        bool positive = false;
+        bool negative = false;
+        for (auto i : tempPoints) {
+          Vector3 vertex = m_vertexData[i];
+          double distance = face.m_P.m_N.dotProduct(vertex) + face.m_P.m_D;
+          if (distance > m_epsilon) {
+              positive = true;
+          } else if (distance < -m_epsilon) {
+              negative = true;
+          }
+
+          // If we find both positive and negative distances, the polyhedron is not convex
+          if (positive && negative) {
```

Comment:
should there be any positive distances for a valid convex hulll?

Reply:
Yeah, so the way I'm checking it is if all the other points that are a part of the convex hull, don't lie on the same side of the face it's not a convex hull. I wrote this to account for incorrect orientation of the normal, since I am testing it after each iteration, and I wasn't sure whether the normals were oriented correctly after each step or just at the last step.

Although, now that you've pointed it out, I think I should have tried checking only for the negatives first in case the normals are oriented correctly, it will be a more sound check. I will check the code to verify the normals aren't rearranged later and if it works I'll make the changes. Thanks for pointing it out